main.go

```diff
@@ -15,32 +16,38 @@ import (
 	"github.com/fsnotify/fsnotify"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/rs/zerolog/log"
 	"github.com/sethgrid/pester"
-
-	"github.com/prometheus/client_golang/prometheus"
-
-	"github.com/ericchiang/k8s"
-	corev1 "github.com/ericchiang/k8s/apis/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 )
 
-const annotationGCPServiceAccount string = "estafette.io/gcp-service-account"
-const annotationGCPServiceAccountName string = "estafette.io/gcp-service-account-name"
-const annotationGCPServiceAccountFilename string = "estafette.io/gcp-service-account-filename"
-const annotationGCPServiceAccountDisableKeyRotation string = "estafette.io/gcp-service-account-disable-key-rotation"
-const annotationGCPServiceAccountPermissions string = "estafette.io/gcp-service-account-permissions"
-const annotationGCPServiceAccountState string = "estafette.io/gcp-service-account-state"
+const (
+	annotationGCPServiceAccount                   string = "estafette.io/gcp-service-account"
+	annotationGCPServiceAccountName               string = "estafette.io/gcp-service-account-name"
+	annotationGCPServiceAccountFilename           string = "estafette.io/gcp-service-account-filename"
+	annotationGCPServiceAccountDisableKeyRotation string = "estafette.io/gcp-service-account-disable-key-rotation"
+	annotationGCPServiceAccountPermissions        string = "estafette.io/gcp-service-account-permissions"
+	annotationGCPServiceAccountState              string = "estafette.io/gcp-service-account-state"
+
+	annotationWorldloadIdentity string = "iam.gke.io/gcp-service-account"
+)
 
 // GCPServiceAccountState represents the state of the secret with respect to GCP service accounts
 type GCPServiceAccountState struct {
-	Enabled                string                        `json:"enabled"`
-	Name                   string                        `json:"name"`
-	Filename               string                        `json:"filename,omitempty"`
-	DisableKeyRotation     bool                          `json:"disableKeyRotation"`
-	FullServiceAccountName string                        `json:"fullServiceAccountName"`
-	Permissions            []GCPServiceAccountPermission `json:"permissions,omitempty"`
-	LastRenewed            string                        `json:"lastRenewed"`
-	LastAttempt            string                        `json:"lastAttempt"`
+	Enabled                 string                        `json:"enabled"`
+	Name                    string                        `json:"name"`
+	Filename                string                        `json:"filename,omitempty"`
+	DisableKeyRotation      bool                          `json:"disableKeyRotation"`
+	FullServiceAccountName  string                        `json:"fullServiceAccountName"`
+	fullServiceAccountEmail string                        `json:"fullServiceAccountEmail"`
```

Comment:
Please fix casing so it starts with a capital, otherwise unmarshalling doesn't work (public properties start with a capital and).